service.subtitles.legendasdivx/service.py

```diff
@@ -293,293 +308,288 @@ def append_subtitle(item):
-		item = {}
-		item['temp']               = False
-		item['rar']                = False
-		item['year']               = xbmc.getInfoLabel("VideoPlayer.Year")                           # Year
-		item['season']             = str(xbmc.getInfoLabel("VideoPlayer.Season"))                    # Season
-		item['episode']            = str(xbmc.getInfoLabel("VideoPlayer.Episode"))                   # Episode
-		item['tvshow']             = str(xbmc.getInfoLabel("VideoPlayer.TVshowtitle"))   # Show
-		item['title']              = str(xbmc.getInfoLabel("VideoPlayer.OriginalTitle")) # try to get original title
-		item['file_original_path'] = urllib.parse.unquote(xbmc.Player().getPlayingFile())  # Full path of a playing file
-		item['mansearch'] = False
-		item['languages'] = []
-
-		if 'searchstring' in params:
-			item['mansearch'] = True
-			item['mansearchstr'] = urllib.parse.unquote(params['searchstring'])
-
-		for lang in urllib.parse.unquote(params['languages']).split(','):
-			item['languages'].append(xbmc.convertLanguage(lang, xbmc.ISO_639_2))
-
-
-		if "s" in item['episode'].lower():
-			# Check if season is "Special"
-			item['season'] = "0"
-			item['episode'] = item['episode'][-1:]
-
-		if "http" in item['file_original_path']: item['temp'] = True
-
-		elif "rar://" in item['file_original_path']:
-			item['rar'] = True
-			item['file_original_path'] = os.path.dirname(item['file_original_path'][6:])
-
-		elif "stack://" in item['file_original_path']:
-			stackPath = item['file_original_path'].split(" , ")
-			item['file_original_path'] = stackPath[0][8:]
-
-		Search(item)
-
-	elif params['action'] == 'download':
-		# we pickup all our arguments sent from def Search()
-		subs = Download(params["id"], params["filename"])
-		# we can return more than one subtitle for multi CD versions, for now we
-		# are still working out how to handle that in XBMC core
-		for sub in subs:
-			listitem = xbmcgui.ListItem(label=sub)
-			xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=sub, listitem=listitem, isFolder=False)
-
-	# Send end of directory to XBMC
-	xbmcplugin.endOfDirectory(int(sys.argv[1]))
\ No newline at end of file
+    def Search(item):
+        """Called when searching for subtitles from KODI."""
+        #### Do what's needed to get the list of subtitles from service site
+        #### use item["some_property"] that was set earlier
+        #### once done, set xbmcgui.ListItem() below and pass it to xbmcplugin.addDirectoryItem()
+        #### CHECKING FOR ANYTHING IN THE USERNAME AND PASSWORD, IF NULL IT STOPS THE SCRIPT WITH A WARNING
+        username = _addon.getSetting( 'LDuser' )
+        password = _addon.getSetting( 'LDpass' )
+        if username == '' or password == '':
+            xbmcplugin.endOfDirectory(int(sys.argv[1]))
+            if username == '' and password != '': xbmc.executebuiltin(('Notification(%s,%s,%d)' % (_scriptname , _language(32016).encode('utf-8'),5000)))
```

Comment:
Can you please avoid the built-in function here? We have access to this on the python API:
xbmcgui.Dialog().notification()

Reply:
Done.